tracing: the CDB process crash if enable COCKROACH_TRACE_SQL

The header (i.e. the first few lines) of the most recent log file. 
````
I170228 11:06:58.060465 1 util/log/clog.go:999  [config] file created at: 2017/02/28 11:06:58
I170228 11:06:58.060465 1 util/log/clog.go:999  [config] running on machine: gzns-dba-sz-ndbtest-02
I170228 11:06:58.060465 1 util/log/clog.go:999  [config] binary: CockroachDB CCL beta-20170223 (linux amd64, built 2017/02/23 16:30:53, go1.7.5)
I170228 11:06:58.060465 1 util/log/clog.go:999  [config] arguments: [/home/work/setup-cockroachdb/bin//cockroach start --port=26257 --http-port=8080 --store=path=/ssd/newdb,attrs=ssd --join=gzns-dba-sz-ndbtest-01.gzns --join=gzns-dba-sz-ndbtest-02.gzns --join=gzns-dba-sz-ndbtest-03.gzns --join=gzns-dba-sz-ndbtest-04.gzns --cache=24G --attrs=gzns --insecure]
````

Here is the call stack:
````
panic: No span in context? Was resetForNewSQLTxn() called previously? [recovered]
        panic: SELECT timestamp, eventType, targetID, reportingID, info, uniqueID FROM system.eventlog WHERE true ORDER BY timestamp DESC LIMIT $1: No span in context? Was resetForNewSQLTxn() called previously?
        panic: session: unexpected leftover memory: 12190 bytes
goroutine 3338 [running]:
panic(0x196c140, 0xc420ed2150)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/sql/mon.(*MemoryMonitor).Stop(0xc421f819a0, 0x7f4ff95844a8, 0xc422b860c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/mon/mem_usage.go:339 +0x381
github.com/cockroachdb/cockroach/pkg/sql.(*Session).Finish(0xc421f80000, 0xc420444000)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/session.go:178 +0x99
panic(0x19ba540, 0xc420ed2120)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements.func1(0xc421b34240, 0x83)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:484 +0x149
panic(0x196c140, 0xc420ed20e0)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/pkg/sql.(*txnState).finishSQLTxn(0xc421f80068, 0x7f4ff95844a8, 0xc422b860c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/session.go:371 +0x26f
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420444000, 0xc421f80000, 0xc421b34240, 0x83, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:742 +0xd45
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements(0xc420444000, 0xc421f80000, 0xc421b34240, 0x83, 0xc422cfc758, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:490 +0x11f
github.com/cockroachdb/cockroach/pkg/server.(*adminServer).Events(0xc4204f2000, 0x7f4ff95843c8, 0xc420d3ad20, 0xc420ed4720, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/admin.go:706 +0x473
````

--- a/pkg/sql/session.go
+++ b/pkg/sql/session.go
@@ -301,13 +301,11 @@ func (ts *txnState) resetForNewSQLTxn(e *Executor, s *Session) {
 	// TODO(andrei): figure out how to close these spans on server shutdown? Ties
 	// into a larger discussion about how to drain SQL and rollback open txns.
 	ctx := s.context
-	var sp opentracing.Span
 	if traceSQL {
 		var err error
 		ctx, ts.trace, err = tracing.StartSnowballTrace(ctx, "traceSQL")
 		if err != nil {
-			log.Warningf(ctx, "unable to create snowball tracer: %s", err)
-			return
+			log.Fatalf(ctx, "unable to create snowball tracer: %s", err)
 		}
 	} else if traceSQLFor7881 {
 		var err error
@@ -316,6 +314,7 @@ func (ts *txnState) resetForNewSQLTxn(e *Executor, s *Session) {
 			log.Fatalf(ctx, "couldn't create a tracer for debugging #7881: %s", err)
 		}
 	} else {
+		var sp opentracing.Span
 		if parentSp := opentracing.SpanFromContext(ctx); parentSp != nil {
 			// Create a child span for this SQL txn.
 			tracer := parentSp.Tracer()
@@ -325,10 +324,11 @@ func (ts *txnState) resetForNewSQLTxn(e *Executor, s *Session) {
 			tracer := e.cfg.AmbientCtx.Tracer
 			sp = tracer.StartSpan("sql txn")
 		}
+		// Put the new span in the context.
+		ctx = opentracing.ContextWithSpan(ctx, sp)
 	}
-	// Put the new span in the context.
-	ts.sp = sp
-	ctx = opentracing.ContextWithSpan(ctx, sp)
+
+	ts.sp = opentracing.SpanFromContext(ctx)
 	ts.Ctx = ctx
 
 	ts.mon.Start(ctx, &s.mon, mon.BoundAccount{})

